be/src/util/arrow/row_batch.cpp

```diff
@@ -211,47 +211,52 @@ class FromRowBatchConverter : public arrow::TypeVisitor {
         size_t num_rows = _batch.num_rows();
         builder.Reserve(num_rows);
         for (size_t i = 0; i < num_rows; ++i) {
-            auto cell_ptr = _cur_slot_ref->get_slot(_batch.get_row(i));
-            PrimitiveType primitive_type = _cur_slot_ref->type().type;
-            switch (primitive_type) {
-                case TYPE_VARCHAR:
-                case TYPE_CHAR:
-                case TYPE_HLL: {
-                    const StringValue* string_val = (const StringValue*)(cell_ptr);
-                    if (string_val == nullptr) {
-                        ARROW_RETURN_NOT_OK(builder.AppendNull());
-                    } else {
-                        if (string_val->len == 0) {
-                            // 0x01 is a magic num, not usefull actually, just for present ""
-                            //char* tmp_val = reinterpret_cast<char*>(0x01);
-                            ARROW_RETURN_NOT_OK(builder.Append(""));        
+            bool is_null = _cur_slot_ref->is_null_bit_set(_batch.get_row(i));
+            if (is_null) {
+                ARROW_RETURN_NOT_OK(builder.AppendNull());
```

Comment:
```suggestion
                ARROW_RETURN_NOT_OK(builder.AppendNull());
                continue;
```
then you can remove` else {`